### DISCUSSION TITLE
Horizontal convection with `VerticallyStretchedRectilinearGrid`

### DISCUSSION THREAD

**Discussion by navidcy** [CliMA/Oceananigans.jl#2167] [General]
The horizontal convection example with a boundary layer at the top is ideal for a `VerticallyStretchedRectilinearGrid`. After #1865, we can now have `VerticallyStretchedRectilinearGrid` with Flat dimensions. Ideally we'd like to change the example to use a `VerticallyStretchedRectilinearGrid`.

**Comment by tomchor** [comment]
> The horizontal convection example with a boundary layer at the top is ideal for a VerticallyStretchedRectilinearGrid

I'm not sure I agree with this statement though. Just to be clear: I'm not saying I disagree, it's just not obvious to me that that's the case.

From the videos in the example the viscous boundary layer seems pretty well-resolved to me (unless you're planning on changing the Pr (or Ra) number?). Plus the turbulence pretty much covers the whole domain, so I don't think we'd want a very coarse grid at the bottom.

Is there a metric to look at to gauge the "resolvedness" of the viscous boundary layer?

**Comment by francispoulin** [comment]
From looking at the animation, the vorticity doesn't show much is going on at the surface, but the buoyancy dissipation is strongest at the surface and changes rapidly. If we had a denser grid near the surface then this could be better resolved.

How many grid points do we have in the boundary layer that develops at the top?  By boundary layer I mean the region of high buoyancy dissiaption?  

I would think that we should have at the very least 5 points per boundary layer but certainly 10 would be better.

**Comment by navidcy** [comment]
Hm... I see both your points.

The boundary layer is indeed pronounced in the buoyancy or buoyancy dissipation. But indeed for Ra=1e8 the boundary layer is most probably well-resolved.

The boundary layer thickness, however, scales with Ra^{-1/5} or Ra^{-1/4} (depending on the flow regime)... Thus, for the Ra value in the example you are right: no modification is needed. But if one wants to take this up to higher Ra values then higher vertical resolution near the top will help.

Perhaps we close this issue (as, I agree, is not really an _issue_ of the code). Or convert to discussion?

**Comment by francispoulin** [comment]
I am fine with closing the issue but any idea how many grid points we have in the boundary layer?

**Comment by navidcy** [comment]
From eye balling the buoyancy dissipation panel I'd say the boundary layer is 0.05H which means 3-4 grid points.

**Comment by francispoulin** [comment]
Thanks @navidcy .  So it's doing okay but not extremely well resolved.  Going back to your thesis in this issue, I think we could resolve the boundary layer dynamics much better with a stretched grid, and therefore get a more accurate picture of where the action starts.  

I think it would be fun to see this done in a stretched grid but I suppose I can try it sometime and see what I find.

**Comment by glwagner** [comment]
I agree with converting this issue to a discussion!
